src/filesystem/wal_reader_node.cc

```diff
@@ -1,17 +1,36 @@
 #include "filesystem/wal_reader_node.h"
 // third party
+#include <core/reactor.hh>
 // smf
 #include "log.h"
 
 namespace smf {
-wal_reader_node::wal_reader_node(sstring _filename, reader_stats *s)
-  : filename(_filename), rstats_(DTHROW_IFNULL(s)) {}
+wal_reader_node::wal_reader_node(uint64_t epoch,
+                                 sstring _filename,
+                                 reader_stats *s)
+  : starting_epoch(epoch), filename(_filename), rstats_(DTHROW_IFNULL(s)) {}
 wal_reader_node::~wal_reader_node() {}
 
+future<> wal_reader_node::close() { return io_->close(); }
 
-future<> wal_reader_node::close() { return make_ready_future<>(); }
-future<> wal_reader_node::open() { return make_ready_future<>(); }
-future<wal_opts::maybe_buffer> wal_reader_node::get(uint64_t epoch) {
+future<> wal_reader_node::open() {
+  return open_file_dma(filename, open_flags::ro).then([this](file f) {
+    auto shared_f = make_lw_shared<file>(std::move(f));
+    return shared_f->size().then([shared_f, this](uint64_t size) {
```

Comment:
This call should not happen here. THis call should happen only on the lazy_file which is the thing that it needs. 

figure out if the right abstraction is a file or a name for the handle.